src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -509,6 +510,23 @@ public bool PropertyNameCaseInsensitive
             }
         }
 
+        /// <summary>
+        /// Specifies that only members declared at the level of the supplied type's hierarchy should be considered.
+        /// Inherited members are not considered.
+        /// </summary>
+        public bool OnlyDeclaredPropertiesOnInterfaces
```

Comment:
All new public APIs must follow the API review process before being added: 

https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md

To be honest though, I don't think should be made configurable, we should always serialize properties from base interfaces like we do with classes.

Reply:
I only thought cause it was mentioned here:
https://github.com/dotnet/runtime/issues/41749#issuecomment-771885274

If we remove the Configuration option, we also do not need the API Review process.